back/commons/lib/in_array.go

```diff
@@ -0,0 +1,19 @@
+package lib
+
+import (
+	"reflect"
+)
+
+func InArray(val any, array any) (index int) {
+	values := reflect.ValueOf(array)
+
+	if reflect.TypeOf(array).Kind() == reflect.Slice || values.Len() > 0 {
+		for i := 0; i < values.Len(); i++ {
+			if reflect.DeepEqual(val, values.Index(i).Interface()) {
+				return i
+			}
+		}
+	}
+
+	return -1
+}
```

Comment:
Cette version n'est pas plus simple ?


```suggestion
func InArray[T comparable](val T, array []T) int {
    for i, v := range array {
        if v == val {
            return i
        }
    }
    return -1
}
```